pkg/pool-manager/pool.go

```diff
@@ -176,3 +191,41 @@ func getNextMac(currentMac net.HardwareAddr) net.HardwareAddr {
 
 	return currentMac
 }
+
+// Checks if the namespace of an instance is opted in for kubemacpool
+func (p *PoolManager) isInstanceOptedIn(namespaceName, labelSelector, labelValue string) (bool, error) {
+	log.V(3).Info("checking for label in namespace", "webhook labelSelector", labelSelector, "webhook labelValue", labelValue)
+	isNamespaceOptedIn := false
+
+	ns, err := p.kubeClient.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
+	if err != nil {
+		return isNamespaceOptedIn, err
+	}
+
+	log.V(3).Info("namespaceName Labels", "namespaceName", namespaceName, "Labels", ns.GetLabels())
+	if value := ns.GetLabels()[labelSelector]; labelValue == value {
```

Comment:
This will panic if labelSelector is not there, can you do `value, ok := ...` and then `ok && labelValue == value` so we prevent panics ?

Reply:
thought of this but ns.[GetLabels](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/meta.go#L54)() doesn't return error. what would you suggest?

Reply:
I don't mean error I mean golang maps, to prevent panic you have to return also the boolean flag that denotes key presence
```golang
if val, ok := dict["foo"]; ok {
    //do something here
}
```

Reply:
got you.